commons/protocol/src/main/scala/sbt/protocol/Protocol.scala

```diff
@@ -71,9 +73,71 @@ sealed trait ExecutionEngineEvent extends Event
 //              Requests (Reactive API)
 // ------------------------------------------
 
-final case class ClientInfo(uuid: String, configName: String, humanReadableName: String)
+// "protocolVersion" doesn't have semantics like major.minor;
+// versions are supposed to be backward compatible always (if they
+// weren't we'd need to do something more radical than send this,
+// pre-connection, such as change the sbt server launch
+// procedure). It's probably easier to use FeatureTag for most
+// "negotiation" scenarios, too. But you could also have a rule
+// like "in protocol version 3, if both sides are >= 3 you do
+// xyz."
+sealed abstract class ProtocolVersion protected (val name: String) {
```

Comment:
I think later is fine... 
(but we aren't actually locking down the *ABI* of this package I don't think, just the JSON? so rather than make a new protocol package we can add fields to these case classes and bump the protocol version)